```
[Error] main.cpp@105,26: too many arguments to function call, expected 0, have 1
[Error] main.cpp@107,23: too many arguments to function call, expected 0, have 1
[Error] main.cpp@109,23: too many arguments to function call, expected 0, have 1
[ERROR] ./main.cpp:105:26: error: too many arguments to function call, expected 0, have 1
    wifi->get_ip_address(&a);
    ~~~~~~~~~~~~~~~~~~~~ ^~
./mbed-os/features/netsocket/NetworkInterface.h:108:5: note: 'get_ip_address' declared here
    virtual const char *get_ip_address();
    ^
./main.cpp:107:23: error: too many arguments to function call, expected 0, have 1
    wifi->get_netmask(&a);
    ~~~~~~~~~~~~~~~~~ ^~
./mbed-os/features/netsocket/NetworkInterface.h:115:5: note: 'get_netmask' declared here
    virtual const char *get_netmask();
    ^
./main.cpp:109:23: error: too many arguments to function call, expected 0, have 1
    wifi->get_gateway(&a);
    ~~~~~~~~~~~~~~~~~ ^~
./mbed-os/features/netsocket/NetworkInterface.h:122:5: note: 'get_gateway' declared here
    virtual const char *get_gateway();
```

@AnttiKauppila Is this caused by master changes + example - not related to this PR but rather the feature branch?

--- a/UNITTESTS/features/lorawan/loramaccommand/Test_LoRaMacCommand.cpp
+++ b/UNITTESTS/features/lorawan/loramaccommand/Test_LoRaMacCommand.cpp
@@ -78,6 +78,10 @@ TEST_F(Test_LoRaMacCommand, parse_mac_commands_to_repeat)
 
     object->parse_mac_commands_to_repeat();
 
+    buf[0] = 1;
+    buf[1] = 1;
+    EXPECT_TRUE(object->process_mac_commands(buf, 0, 2, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_OK);
+
     buf[0] = 2;
     buf[1] = 16;
     buf[2] = 32;
@@ -122,6 +126,47 @@ TEST_F(Test_LoRaMacCommand, parse_mac_commands_to_repeat)
 
     buf[0] = 10;
     buf[1] = 4;
+    EXPECT_TRUE(object->process_mac_commands(buf, 0, 4, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_OK);
+
+    buf[0] = 11;
+    buf[1] = 1;
+    EXPECT_TRUE(object->process_mac_commands(buf, 0, 2, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_OK);
+
+    buf[0] = 12;
+    buf[1] = 2;
+    EXPECT_TRUE(object->process_mac_commands(buf, 0, 2, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_OK);
+
+    buf[0] = 13;
+    buf[1] = 2;
+    buf[2] = 2;
+    buf[3] = 2;
+    buf[4] = 2;
+    buf[5] = 2;
+    EXPECT_TRUE(object->process_mac_commands(buf, 0, 6, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_OK);
+
+    buf[0] = 14;
+    buf[1] = 2;
+    buf[2] = 2;
+    EXPECT_TRUE(object->process_mac_commands(buf, 0, 3, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_OK);
+
+    buf[0] = 15;
+    buf[1] = 2;
+    EXPECT_TRUE(object->process_mac_commands(buf, 0, 2, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_OK);
+
+    buf[0] = 16;
+    EXPECT_TRUE(object->process_mac_commands(buf, 0, 1, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_OK);
+
+    buf[0] = 17;
+    buf[1] = 4;
+    buf[2] = 2;
+    buf[3] = 2;
+    buf[4] = 2;
+    EXPECT_TRUE(object->process_mac_commands(buf, 0, 5, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_OK);
+
+    buf[0] = 19;
+    buf[1] = 4;
+    buf[2] = 2;
+    buf[3] = 2;
     EXPECT_TRUE(object->process_mac_commands(buf, 0, 4, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_OK);
 
     object->parse_mac_commands_to_repeat();
@@ -215,6 +260,11 @@ TEST_F(Test_LoRaMacCommand, process_mac_commands)
     uint8_t buf[20];
     EXPECT_TRUE(object->process_mac_commands(NULL, 0, 0, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_OK);
 
+    buf[0] = 1;
+    buf[1] = 1;
+    EXPECT_TRUE(object->process_mac_commands(buf, 0, 2, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_OK);
+
+    object->clear_command_buffer();
     buf[0] = 2;
     buf[1] = 16;
     buf[2] = 32;
@@ -330,6 +380,83 @@ TEST_F(Test_LoRaMacCommand, process_mac_commands)
     for (int i = 0; i < 64; i++) {
         EXPECT_TRUE(object->process_mac_commands(buf, 0, 4, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_OK);
         LoRaPHY_stub::bool_counter = 0;
+    }
+    EXPECT_TRUE(object->process_mac_commands(buf, 0, 4, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_LENGTH_ERROR);
+
+    object->clear_command_buffer();
+    buf[0] = 11;
+    buf[1] = 1;
+    EXPECT_TRUE(object->process_mac_commands(buf, 0, 2, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_OK);
+
+    object->clear_command_buffer();
+    buf[0] = 12;
+    buf[1] = 2;
+    EXPECT_TRUE(object->process_mac_commands(buf, 0, 2, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_OK);
+
+    //overflow add_adr_param_setup_ans
+    object->clear_command_buffer();
+    for (int i = 0; i < 128; i++) {
+        EXPECT_TRUE(object->process_mac_commands(buf, 0, 2, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_OK);
+    }
+    EXPECT_TRUE(object->process_mac_commands(buf, 0, 2, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_LENGTH_ERROR);
+
+    object->clear_command_buffer();
+    buf[0] = 13;
+    buf[1] = 2;
+    buf[2] = 2;
+    buf[3] = 2;
+    buf[4] = 2;
+    buf[5] = 2;
+    EXPECT_TRUE(object->process_mac_commands(buf, 0, 6, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_OK);
+
+    object->clear_command_buffer();
+    buf[0] = 14;
+    buf[1] = 2;
+    buf[2] = 2;
+    EXPECT_TRUE(object->process_mac_commands(buf, 0, 3, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_OK);
+
+    object->clear_command_buffer();
+    buf[0] = 15;
+    buf[1] = 2;
+    EXPECT_TRUE(object->process_mac_commands(buf, 0, 2, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_OK);
+
+    //overflow add_rejoin_param_setup_ans
+    object->clear_command_buffer();
+    for (int i = 0; i < 64; i++) {
+        EXPECT_TRUE(object->process_mac_commands(buf, 0, 2, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_OK);
+    }
+    EXPECT_TRUE(object->process_mac_commands(buf, 0, 2, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_LENGTH_ERROR);
+
+    object->clear_command_buffer();
+    buf[0] = 16;
+    EXPECT_TRUE(object->process_mac_commands(buf, 0, 1, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_OK);
+
+    object->clear_command_buffer();
+    buf[0] = 17;
+    buf[1] = 4;
+    buf[2] = 2;
+    buf[3] = 2;
+    buf[4] = 2;
+    EXPECT_TRUE(object->process_mac_commands(buf, 0, 5, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_OK);
+
+    //Overflow add_ping_slot_channel_ans
+    object->clear_command_buffer();
+    for (int i = 0; i < 64; i++) {
+        EXPECT_TRUE(object->process_mac_commands(buf, 0, 5, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_OK);
+    }
+    EXPECT_TRUE(object->process_mac_commands(buf, 0, 5, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_LENGTH_ERROR);
+
+    object->clear_command_buffer();
+    buf[0] = 19;
+    buf[1] = 4;
+    buf[2] = 2;
+    buf[3] = 2;
+    EXPECT_TRUE(object->process_mac_commands(buf, 0, 4, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_OK);
+
+    //Overflow add_beacon_freq_ans
+    object->clear_command_buffer();
+    for (int i = 0; i < 64; i++) {
+        EXPECT_TRUE(object->process_mac_commands(buf, 0, 4, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_OK);
     }
     EXPECT_TRUE(object->process_mac_commands(buf, 0, 4, 0, params, phy, my_cb, RX_SLOT_WIN_1) == LORAWAN_STATUS_LENGTH_ERROR);
 

--- a/features/lorawan/lorastack/mac/LoRaMacCommand.cpp
+++ b/features/lorawan/lorastack/mac/LoRaMacCommand.cpp
@@ -67,8 +67,9 @@ void LoRaMacCommand::parse_mac_commands_to_repeat()
     for (i = 0; i < mac_cmd_buf_idx; i++) {
         switch (mac_cmd_buffer[i]) {
             // STICKY
-            case MOTE_MAC_DL_CHANNEL_ANS:
+            case MOTE_MAC_RESET_IND:
             case MOTE_MAC_RX_PARAM_SETUP_ANS:
+            case MOTE_MAC_DL_CHANNEL_ANS:
             case MOTE_MAC_REKEY_IND: { // 1 byte payload
                 mac_cmd_buffer_to_repeat[cmd_cnt++] = mac_cmd_buffer[i++];
                 mac_cmd_buffer_to_repeat[cmd_cnt++] = mac_cmd_buffer[i];
@@ -85,13 +86,16 @@ void LoRaMacCommand::parse_mac_commands_to_repeat()
             }
             case MOTE_MAC_LINK_ADR_ANS:
             case MOTE_MAC_NEW_CHANNEL_ANS:
-            case MOTE_MAC_PING_SLOT_INFO_REQ: { // 1 byte payload
+            case MOTE_MAC_REJOIN_PARAM_SETUP_ANS:
+            case MOTE_MAC_PING_SLOT_INFO_REQ:
+            case MOTE_MAC_PING_SLOT_CHANNEL_ANS: { // 1 byte payload
                 i++;
                 break;
             }
-            case MOTE_MAC_TX_PARAM_SETUP_ANS:
-            case MOTE_MAC_DUTY_CYCLE_ANS:
             case MOTE_MAC_LINK_CHECK_REQ:
+            case MOTE_MAC_DUTY_CYCLE_ANS:
+            case MOTE_MAC_TX_PARAM_SETUP_ANS:
+            case MOTE_MAC_ADR_PARAM_SETUP_ANS:
             case MOTE_MAC_DEVICE_TIME_REQ: { // 0 byte payload
                 break;
             }
@@ -366,7 +370,7 @@ lorawan_status_t LoRaMacCommand::process_mac_commands(const uint8_t *payload, ui
                 uint32_t max_time = 1 << time;
 
                 uint8_t time_available = lora_phy.update_rejoin_params(max_time, max_count);
-                add_rejoin_param_setup_ans(time_available);
+                ret_value = add_rejoin_param_setup_ans(time_available);
             }
             break;
             case SRV_MAC_DEVICE_MODE_CONF: {
@@ -604,6 +608,7 @@ lorawan_status_t LoRaMacCommand::add_adr_param_setup_ans()
     lorawan_status_t ret = LORAWAN_STATUS_LENGTH_ERROR;
     if (cmd_buffer_remaining() > 0) {
         mac_cmd_buffer[mac_cmd_buf_idx++] = MOTE_MAC_ADR_PARAM_SETUP_ANS;
+        ret = LORAWAN_STATUS_OK;
     }
     return ret;
 }
@@ -611,9 +616,10 @@ lorawan_status_t LoRaMacCommand::add_adr_param_setup_ans()
 lorawan_status_t LoRaMacCommand::add_rejoin_param_setup_ans(uint8_t status)
 {
     lorawan_status_t ret = LORAWAN_STATUS_LENGTH_ERROR;
-    if (cmd_buffer_remaining() > 0) {
+    if (cmd_buffer_remaining() > 1) {
         mac_cmd_buffer[mac_cmd_buf_idx++] = MOTE_MAC_REJOIN_PARAM_SETUP_ANS;
         mac_cmd_buffer[mac_cmd_buf_idx++] = status & 0x01;
+        ret = LORAWAN_STATUS_OK;
     }
     return ret;
 }
@@ -623,9 +629,10 @@ lorawan_status_t LoRaMacCommand::add_ping_slot_info_req(uint8_t periodicity)
     lorawan_status_t ret = LORAWAN_STATUS_LENGTH_ERROR;
 
     if ((periodicity & 0x07) == periodicity) {
-        if (cmd_buffer_remaining() > 0) {
+        if (cmd_buffer_remaining() > 1) {
             mac_cmd_buffer[mac_cmd_buf_idx++] = MOTE_MAC_PING_SLOT_INFO_REQ;
             mac_cmd_buffer[mac_cmd_buf_idx++] = periodicity;
+            ret = LORAWAN_STATUS_OK;
         }
     } else {
         ret = LORAWAN_STATUS_PARAMETER_INVALID;
@@ -636,9 +643,10 @@ lorawan_status_t LoRaMacCommand::add_ping_slot_info_req(uint8_t periodicity)
 lorawan_status_t LoRaMacCommand::add_ping_slot_channel_ans(uint8_t status)
 {
     lorawan_status_t ret = LORAWAN_STATUS_LENGTH_ERROR;
-    if (cmd_buffer_remaining() > 0) {
+    if (cmd_buffer_remaining() > 1) {
         mac_cmd_buffer[mac_cmd_buf_idx++] = MOTE_MAC_PING_SLOT_CHANNEL_ANS;
         mac_cmd_buffer[mac_cmd_buf_idx++] = status & 0x03;
+        ret = LORAWAN_STATUS_OK;
     }
     return ret;
 }
@@ -647,9 +655,10 @@ lorawan_status_t LoRaMacCommand::add_beacon_freq_ans(uint8_t status)
 {
     lorawan_status_t ret = LORAWAN_STATUS_LENGTH_ERROR;
 
-    if (cmd_buffer_remaining() > 0) {
-        mac_cmd_buffer[mac_cmd_buf_idx++] = MOTE_MAC_PING_SLOT_CHANNEL_ANS;
+    if (cmd_buffer_remaining() > 1) {
+        mac_cmd_buffer[mac_cmd_buf_idx++] = MOTE_MAC_BEACON_FREQ_ANS;
         mac_cmd_buffer[mac_cmd_buf_idx++] = status & 0x01;
+        ret = LORAWAN_STATUS_OK;
     }
     return ret;
 }

